pkg/monitor/monitor/util.go

```diff
@@ -1003,3 +1033,131 @@ func getMonitorVolumeClaims(monitor *v1alpha1.TidbMonitor) []core.PersistentVolu
+		}
+	}
+
+	container := core.Container{
+		Name:      "thanos-sidecar",
+		Image:     fmt.Sprintf("%s:%s", thanos.BaseImage, thanos.Version),
+		Resources: controller.ContainerResource(thanos.ResourceRequirements),
+		Args:      thanosArgs,
+		Env: []core.EnvVar{
+			{
+				Name: "POD_IP",
+				ValueFrom: &core.EnvVarSource{
+					FieldRef: &core.ObjectFieldSelector{
+						FieldPath: "status.podIP",
+					},
+				},
+			},
+			{
+				Name: "POD_NAME",
+				ValueFrom: &core.EnvVarSource{
+					FieldRef: &core.ObjectFieldSelector{FieldPath: "metadata.name"},
+				},
+			},
+		},
+		Ports: []core.ContainerPort{
+			{
+				Name:          "http",
+				ContainerPort: 10902,
+				Protocol:      "TCP",
+			},
+			{
+				Name:          "grpc",
+				ContainerPort: 10901,
+				Protocol:      "TCP",
+			},
+		},
+	}
+	if thanos.ObjectStorageConfig != nil || thanos.ObjectStorageConfigFile != nil {
+		if thanos.ObjectStorageConfig != nil {
+			container.Args = append(container.Args, "--objstore.config=$(OBJSTORE_CONFIG)")
+			container.Env = append(container.Env, core.EnvVar{
+				Name: "OBJSTORE_CONFIG",
+				ValueFrom: &core.EnvVarSource{
+					SecretKeyRef: thanos.ObjectStorageConfig,
+				},
+			})
+		} else {
+			if thanos.ObjectStorageConfigFile != nil {
+				container.Args = append(container.Args, "--objstore.config-file="+*thanos.ObjectStorageConfigFile)
+			}
+		}
+		storageDir := "/data/prometheus"
+		container.Args = append(container.Args, fmt.Sprintf("--tsdb.path=%s", storageDir))
+		container.VolumeMounts = append(
+			container.VolumeMounts,
+			core.VolumeMount{
+				Name:      v1alpha1.TidbMonitorMemberType.String(),
+				MountPath: "/data",
+			},
+		)
+	}
+
+	if thanos.TracingConfig != nil {
+		container.Args = append(container.Args, "--tracing.config=$(TRACING_CONFIG)")
+		container.Env = append(container.Env, core.EnvVar{
+			Name: "TRACING_CONFIG",
+			ValueFrom: &core.EnvVarSource{
+				SecretKeyRef: thanos.TracingConfig,
+			},
+		})
+	}
+
+	if thanos.LogLevel != "" {
+		container.Args = append(container.Args, "--log.level="+thanos.LogLevel)
+	}
+	if thanos.LogFormat != "" {
+		container.Args = append(container.Args, "--log.format="+thanos.LogFormat)
+	}
+
+	if thanos.MinTime != "" {
+		container.Args = append(container.Args, "--min-time="+thanos.MinTime)
+	}
+	return container
+}
+
+func buildExternalLabels(monitor *v1alpha1.TidbMonitor) model.LabelSet {
+	m := model.LabelSet{}
+	// Use defaultReplicaExternalLabelName constant by default if field is missing.
+	// Do not add external label if field is set to empty string.
+	replicaExternalLabelName := defaultReplicaExternalLabelName
+	if monitor.Spec.ReplicaExternalLabelName != nil {
+		if *monitor.Spec.ReplicaExternalLabelName != "" {
+			replicaExternalLabelName = *monitor.Spec.ReplicaExternalLabelName
+		} else {
+			replicaExternalLabelName = ""
+		}
+	}
+	if replicaExternalLabelName != "" {
+		m[model.LabelName(replicaExternalLabelName)] = "$(POD_NAME)"
```

Comment:
It is needed to append other parts as the label value? like namespace?

Reply:
Does adding namespace guarantee uniqueness?

Reply:
yes, it seems this label value should be unique.

Reply:
I don't think the Ns tag is needed. If this is to add ns to the environment variable, we just provide a solution. Users can also customize these labels.

Reply:
@DanielZhangQD WDYT

Reply:
I think it's OK to add the namespace to the value.